core/key.go

```diff
@@ -106,6 +106,19 @@ type CollectionSchemaVersionKey struct {
 
 var _ Key = (*CollectionSchemaVersionKey)(nil)
 
+// SchemaHistoryKey holds the pathway through the schema version history for
+// any given schema.
+//
+// The keys point to the schema version id of the next version of the schema.
+// If a SchemaHistoryKey does not exist for a given SchemaVersionID it means
+// that that SchemaVersionID is for the latest version.
+type SchemaHistoryKey struct {
+	SchemaID             string
+	PriorSchemaVersionId string
```

Comment:
nitpick: Slight preference for `Previous` vs `Prior`, I feel its a little more common and is often used in the programming context in regards to `next` / `prev` (previous).

Reply:
Can do, not a problem

- [ ] Rename prior to previous